lib/zip/entry_set.rb

```diff
@@ -13,8 +13,10 @@ def include?(entry)
       @entry_set.include?(to_key(entry))
     end
 
-    def find_entry(entry)
-      @entry_set[to_key(entry)]
+    def find_entry(entry, case_sensitively = true)
+      return @entry_set[to_key(entry)] if case_sensitively
+      entry = @entry_set.find { |k, _| k.downcase == to_key(entry).downcase }
```

Comment:
@rastarobot, in short. *this* is how you do it: https://github.com/rubyzip/rubyzip/compare/master...weshatheleopard:master (sorry, I couldn't run tests because `simplecov` refuses to install on my home box, and I have no time to figure out why).